google/cloud/aiplatform/prediction/local_model.py

```diff
@@ -118,35 +108,43 @@ def create(
                 no impact on whether the port is actually exposed, any port listening on
                 the default "0.0.0.0" address inside a container will be accessible from
                 the network.
-
-        Returns:
-            local model: Instantiated representation of the local model.
         """
-        env = None
-        ports = None
-
-        if serving_container_environment_variables:
-            env = [
-                gca_env_var_compat.EnvVar(name=str(key), value=str(value))
-                for key, value in serving_container_environment_variables.items()
-            ]
-        if serving_container_ports:
-            ports = [
-                gca_model_compat.Port(container_port=port)
-                for port in serving_container_ports
-            ]
+        if serving_container_spec:
+            if not serving_container_spec.image_uri:
+                raise ValueError(
+                    "Image uri is required for the serving container spec to initialize a LocalModel instance."
+                )
 
-        container_spec = gca_model_compat.ModelContainerSpec(
-            image_uri=serving_container_image_uri,
-            command=serving_container_command,
-            args=serving_container_args,
-            env=env,
-            ports=ports,
-            predict_route=serving_container_predict_route,
-            health_route=serving_container_health_route,
-        )
+            self.serving_container_spec = serving_container_spec
+        else:
+            if not serving_container_image_uri:
+                raise ValueError(
+                    "Serving container image uri is required to initialize a LocalModel instance."
+                )
 
-        return cls(container_spec)
+            env = None
+            ports = None
+
+            if serving_container_environment_variables:
+                env = [
+                    gca_env_var_compat.EnvVar(name=str(key), value=str(value))
+                    for key, value in serving_container_environment_variables.items()
+                ]
+            if serving_container_ports:
+                ports = [
+                    gca_model_compat.Port(container_port=port)
+                    for port in serving_container_ports
+                ]
+
+            self.serving_container_spec = gca_model_compat.ModelContainerSpec(
+                image_uri=serving_container_image_uri,
+                command=serving_container_command,
+                args=serving_container_args,
+                env=env,
+                ports=ports,
+                predict_route=serving_container_predict_route,
+                health_route=serving_container_health_route,
+            )
 
     @classmethod
     def create_cpr_model(
```

Comment:
This can be moved into __init__ as well, "create" is a reserved verb

Reply:
I think moving this method into `init` is confusing because `init` will become pretty large, e.g. which parameters go first, which group of parameters should be set together, etc. Could we just rename this method but keep it as a separate method if "create" is a reserved verb?

Reply:
That works

Reply:
Done.